ext/nokogiri/extconf.rb

```diff
@@ -587,6 +591,8 @@ def do_clean
       FileUtils.rm_rf(dir, verbose: true)
     end
 
+    exit!(0) if android?
+
```

Comment:
No matter what I tried, I kept getting:

```
/home/builder/.termux-build/ruby/src/lib/fileutils.rb:1624:in `stat': No such file or directory @ rb_file_s_stat - /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ports/archives/libxml2-2.12.7.tar.xz (Errno::ENOENT)
  from /home/builder/.termux-build/ruby/src/lib/fileutils.rb:1624:in `block in install'
  from /home/builder/.termux-build/ruby/src/lib/fileutils.rb:2469:in `block in fu_each_src_dest'
  from /home/builder/.termux-build/ruby/src/lib/fileutils.rb:2485:in `fu_each_src_dest0'
  from /home/builder/.termux-build/ruby/src/lib/fileutils.rb:2467:in `fu_each_src_dest'
  from /home/builder/.termux-build/ruby/src/lib/fileutils.rb:1623:in `install'
  from /home/builder/.termux-build/ruby/src/tool/rbinstall.rb:190:in `install'
  from /home/builder/.termux-build/ruby/src/tool/rbinstall.rb:881:in `block in extract_files'
  from /home/builder/.termux-build/ruby/src/tool/rbinstall.rb:876:in `each'
  from /home/builder/.termux-build/ruby/src/tool/rbinstall.rb:876:in `extract_files'
  from /home/builder/.termux-build/ruby/src/lib/rubygems/installer.rb:861:in `extract_files'
  from /home/builder/.termux-build/ruby/src/lib/rubygems/installer.rb:328:in `install'
  ...
```

For example, this snippet with multiple safe execution checks (`begin`...`rescue`, `File.directory?`) didn't fix it too:

```rb
    Pathname.glob(root.join("ports", config_static? ? "" : "archives")) do |dir|
      begin
        FileUtils.rm_rf(dir, verbose: true) if File.directory?(dir)
      rescue Errno::ENOENT => e
        puts "Error: #{e.message}"
      end
    end
```

I guess the file/directory exists but the process doesn't have the required permissions to delete those files.

For now I've exited from the program at this checkpoint, but I suppose there could be a better fix.

Reply:
This is not the right solution and breaks the general case. Let's move the conversation back to #3244 or to #3250 if there are problems with that solution.